### PR TITLE
Add initial support for a hygon csv verifier

### PR DESCRIPTION
**Pull request by BaoshunFang**

### CODE REVIEW

**Review by jialez0 — APPROVED** [review]
LGTM.

**Comment by jialez0** [comment]
@BaoshunFang Can you take a look at the CI error?

File: attestation-service/Cargo.toml [hunk 1]
```diff
@@ -50,6 +52,9 @@ tokio.workspace = true
 tonic = { workspace = true, optional = true }
 uuid = { version = "1.1.2", features = ["v4"] }
 x509-parser = { version = "0.14.0", optional = true }
+# TODO: change it to "0.1", once released.                                                                                                                                         
+csv-rs = { git = "https://gitee.com/anolis/csv-rs", rev = "8a90aac", optional = true }
+codicon = { version = "3.0", optional = true }
```
Comment (Xynnn007) [inline attestation-service/Cargo.toml:57]: Hey @BaoshunFang Could you please move the two deps in a dictionarial order?

File: protos/attestation.proto [hunk 2]
```diff
@@ -8,6 +8,7 @@ enum Tee {
     SNP = 2;
     TDX = 3;
     Sample = 4;
+    CSV = 6;
```
Comment (Xynnn007) [inline protos/attestation.proto:11]: is `6` as expected?

**Review by Xynnn007 — APPROVED** [review]
Thanks @BaoshunFang for the PR. Welcome to Confidential Containers and your code looks very great to me! Only some nits.

**Comment by jialez0** [comment]
@BaoshunFang You can run `cargo fmt` to solve the current CI error.

File: protos/attestation.proto (hunk 2 above)
Comment (jialez0) [reply protos/attestation.proto:11]: This is because: https://github.com/confidential-containers/kbs/blob/main/proto/attestation.proto#L5

**Comment by BaoshunFang** [comment]
> 

Done

File: protos/attestation.proto (hunk 2 above)
Comment (BaoshunFang) [reply protos/attestation.proto:11]: Yes, AzSnpVtpm is not added to AS, but to kbs.

File: attestation-service/Cargo.toml (hunk 1 above)
Comment (BaoshunFang) [reply attestation-service/Cargo.toml]: Done

File: attestation-service/src/verifier/csv/mod.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        let tee_evidence = serde_json::from_str::<CsvEvidence>(&attestation.tee_evidence)
+            .context("Deserialize Quote failed.")?;
+
+        verify_report_signature(&tee_evidence)?;
+
+        let report_raw = restore_attestation_report(tee_evidence.attestation_report)?;
+
+        let expected_report_data = calculate_expected_report_data(&nonce, &attestation.tee_pubkey);
+        if report_raw.body.report_data != expected_report_data {
+            return Err(anyhow!("Report Data Mismatch"));
```
[the remark this answers is not in the document]
Comment (BaoshunFang) [reply attestation-service/src/verifier/csv/mod.rs]: Done

**Pull request merged**